java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -832,7 +835,35 @@ public final ColumnVector mergeAndSetValidity(BinaryOp mergeOp, ColumnView... co
       columnViews[i] = columns[i].getNativeView();
     }
 
-    return new ColumnVector(bitwiseMergeAndSetValidity(getNativeView(), columnViews, mergeOp.nativeId));
+    ColumnVector potentialRes = new ColumnVector(bitwiseMergeAndSetValidity(getNativeView(), columnViews, mergeOp.nativeId));
+
+    /**
+     * We need to sanitize the vector to make sure of two things
+     * 1. Nulls set in the new vector don't have non-empty nested types
+     * 2. Nulls in 'this' stay nulls otherwise bad things will happen
+     */
+    ColumnVector finalRes;
+    if (potentialRes.getType().isNestedType() || potentialRes.getType().hasOffsets() && potentialRes.hasNonEmptyNulls()) {
+      // purge non-empty nulls in the vector
+      finalRes = potentialRes.purgeNonEmptyNulls();
+      potentialRes.close();
+    } else {
+      finalRes = potentialRes;
+    }
+
+    // Verify the nulls in the original vector remain null
+    if (getNullCount() > 0) {
```

Comment:
This feels kind of expensive to do. The point of this API originally was just to do something that would be very fast, and it has slowly changed over time to end up being rather expensive.  First off we never have to do this if 

Can we benchmark this to see what the performance is?  Because we are close to burndown for CUDF can you also file a follow on issue to try and find a faster way to do something similar?

If/else and coalesce all provide similar results but we know that they are safe.  Perhaps switching to them would be better.

```
// But switch the and to an or depending on the operation being done.
try (ColumnVector shouldOutputNotBeNull = columns[0].isNotNull()) {
  for (int i = 1; i < columns.length; i++) {
    try (ColumnVector tmpIsNotNull = column[i].isNotNull()) {
      ColumnVector tmpCombined = tmpIdNotNull.and(shouldOutputNotBeNull)
      shouldOutputNotBeNull.close()
      shouldOutputNotBeNull = tmpCombined
    }
  }
  return shouldOutputNotBeNull.ifElse(this, null_scalar)
}
```